crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -192,6 +204,7 @@ impl SoftwareManagerActor {
                 );
                 self.output_sender.send(response.into()).await?;
             }
+            Ok(Some(SoftwareCommand::SoftwareCommandMetadata(_))) => unreachable!(),
```

Comment:
As for the `handle_request` method, I would consider either to do nothing or to return the actual list of supported software types.